CHANGELOG.md

```diff
@@ -170,6 +191,63 @@ function supportsInterface(bytes4 interfaceId) public view virtual override retu
+
+#### ECDSA and MessageHashUtils
+
+The `ECDSA` library now only include functions for recovering the signer from a given signature. Previously it also included utility methods for producing digests to be used in signing or recovery. These utilities were moved to the `MessageHashUtils` library and should be imported if needed:
+
+```diff
+ import {ECDSA} from "@openzeppelin/contracts/utils/cryptography/ECDSA.sol";
++import {MessageHashUtils} from "@openzeppelin/contracts/utils/cryptography/MessageHashUtils.sol";
+
+ contract Verifier {
+   using ECDSA for bytes32;
++  using MessageHashUtils for bytes32;
+ 
+   function _verify(bytes32 data, bytes memory signature, address account) internal pure returns (bool) {
+     return data
+       .toEthSignedMessageHash()
+       .recover(signature) == account;
+   }
+ }
+```
+
+#### Partial transpilation in upgradeable contracts
+
+The upgradeable version of the contracts library used to include a variant suffixed with `Upgradeable` for every contract. These variants are produced via transpilation and mainly include changes for dealing with storage that don't apply to libraries and interfaces.
+
+In favor of their original versions, the library no longer include upgradeable libraries and interfaces. Projects migrating to 5.0 should replace their library and interface imports with their corresponding non-upgradeable version:
+
+```diff
+ // Libraries
+-import {AddressUpgradeable} from '@openzeppelin/contracts-upgradeable/utils/AddressUpgradeable.sol';
++import {Address} from '@openzeppelin/contracts/utils/Address.sol';
+
+ // Interfaces
+-import {IERC20Upgradeable} from '@openzeppelin/contracts-upgradeable/interfaces/IERC20.sol';
++import {IERC20} from '@openzeppelin/contracts/interfaces/IERC20.sol';
+```
+
+#### Offchain Considerations
+
+Some changes may affect offchain systems if they rely on assumptions that are changed along with these new breaking changes. These cases are:
+
+##### Relying on revert strings for processing errors
+
+A concrete example is AccessControl, where it was advised to catch revert reasons using the following regex:
+
+```
+/^AccessControl: account (0x[0-9a-f]{40}) is missing role (0x[0-9a-f]{64})$/
+```
+
+Instead, systems that interact with smart contracts outside of the network should take a look for any dependence on revert strings.
+
+##### Relying on storage locations for retrieving data
+
+After 5.0, both `TransparentUpgradeableProxy` and `BeaconProxy` use immutable variables to avoid unnecessary storage reads on every call. This means that both ERC1967's `BEACON_SLOT` and `ADMIN_SLOT` may no longer reflect the real value if its storage value differs from the immutable one.
```

Comment:
Hm, I would remove this. It does not break with the contracts as we ship them, it requires modification, and that will be something that the developer implementing that modification will have to consider, and not the developer implementing a read from storage.